### ISSUE TITLE
First Aid Kits and Doctor Bag upgrades not working properly for clients

### ISSUE THREAD

**Issue by rockr230** [polarathene/biglobby#43]
If anyone places an upgraded First Aid Kit or Doctor Bag, only the host will be affected by most of the upgrades, such as damage reduction (_"Quick Fix"_ aced), automatic use within a 5 meter radius (_"Uppers"_ aced), and additional Doctor Bag charges (_"Combat Doctor"_ aced). The only client side upgrades available are the First Aid Kit inventory increase effect of _"Uppers"_ aced and Doctor Bag inventory increase effect of _"Combat Doctor"_ basic.
Additionally, when a Doctor Bag with _"Combat Doctor"_ aced is placed, clients only see 2 charges in the bag as if it were a basic version, but can only use one charge. The host can still see/use the full 4 charges of the upgraded bag.

**EDIT**: In case it wasn't clear, the problem with the First Aid Kits for clients is that they cannot use the auto-use feature that the _"Uppers"_ aced skill gives.
The _"Uppers"_ aced skill gives your character additional First Aid Kits (in addition to the additional First Aid Kits from the _"Uppers"_ basic skill) _and_ allows your First Aid Kits to be automatically used within a 5 meter radius.
Basically, what should happen is when someone loses all their health and is within that radius, the First Aid Kit is used on that person. However, the skill then needs a 20 second cool down before being used again. But, when BigLobby is activated, this does not happen for clients.

~~**EDIT**: I think it might be something wrong with the biglobby/pdmod src/mod/network.network_settings file, as I noticed there are no mentions on any first aid kit (around line **1854**) or doctor bag (around line **1149**) upgrades. But, that's just speculation on my part, and I have no idea if I'm interpreting this correctly XP.~~

**Comment by polarathene** [comment]
@rockr230 Line 1149 with the sync possession methods have `biglobby__` prefix versions at the bottom section of the file where I add my changes(originals are left untouched), so those aren't an issue. Line 1854:

```
<message name="first_aid_kit_sync" delivery="ordered" receiver="unit">
	<param type="unit" />
	<param type="float" />
</message>
```

No change needed to be done here as it doesn't refer to the peer id which is the only data that needs to be modified on that file. Same goes for the doctor bag around 1149 you were referring to. Are you certain this is an issue with BigLobby? Have you tested with BigLobby as the only active mod? And without to confirm it only happens with BigLobby mod enabled(with no other mods conflicting with it)?

If you can confirm that, then it's possibly a slight mistake somewhere in my code, though it might not be easy to find and I'm unable to invest the time to do so :( Best guess is if it's a BigLobby method in that networks file that I forgot to add to biglobby_globals which patches the network methods, or it might be related to join method under network folder if Overkill changed the parameters a bit and that information is not sent because my parameters aren't matching.

**Comment by rockr230** [comment]
Tested it with a friend, and confirmed that BigLobby is causing this issue.

**Comment by polarathene** [comment]
@rockr230 If that is the case, I'm unable to fix it due to time. If a community modder wants to help, the information I provided above should help them locate it hopefully. I'm more than happy to answer any questions they might have if you can find someone.

**Comment by rockr230** [comment]
Newest version fixes this!
